src/partials/feedback.html

```diff
@@ -0,0 +1,85 @@
+<!--
+  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to
+  deal in the Software without restriction, including without limitation the
+  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+  sell copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be included in
+  all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+  IN THE SOFTWARE.
+-->
+
+<!-- Determine feedback configuration -->
+{% if config.extra.analytics %}
+  {% set feedback = config.extra.analytics.feedback %}
+{% endif %}
+
+<!-- Determine whether to show feedback -->
+{% if page.meta and page.meta.hide %}
+  {% if "feedback" in page.meta.hide %}
```

Comment:
I guess we need to add a way to set this?

Reply:
That would take me a while because I'm not well versed in site analytics. Here's hoping it is a simple patch to our current implementation of the analytics field in `html_theme_options`.

Reply:
Sorry, I just meant a way to add it to page.meta.hide, e.g. via a `:hide-feedback:` meta option or something --- note the existing code in nav_adapt.py that modifies `page["meta"]["hide"]`

Reply:
I think I found a bug (probably in Sphinx). The metadata keys don't get interpreted as metadata if the have a `-` in them. Rather, they just get interpreted as a field. I have tried all of our supported metadata tags, and they currently don't work. So, all our metadata keys should be changed to `hide_*` (thank goodnes we're still in beta).

I'm still digging, and I'll start rolling back the sphinx version I'm using (currently v5.3.0) to see where it was broken.

Reply:
Well the feedback can be displayed as a demo using a dummy analytics provider (like `G-XXXXXXXXXX`), but it would mean an additional warning in the browser's dev console. I've done this locally to verify that the new `:hide_feedback:` metadata tag works.

I started to look into doing this with my Google account, but apparently all traffic to analytics.google.com domain is blocked by my DNS sinkhole (PiHole) 😅 .

For now, I think I'll just put an example snippet in the docs under the analytics option in customization.rst.

Reply:
> I'm still digging, and I'll start rolling back the sphinx version I'm using (currently v5.3.0) to see where it was broken.

Looks like the JSON domain relies on a module in sphinx that wasn't available in v4.3.0; we should probably update our requirements (or the optional deps for the json domain).

I couldn't find where it was broken. I tried rolling back sphinx (every version after v4.3.0) and docutils (as far back as v0.16), and the metadata tags weren't getting picked up with hyphens in the keys' name (using a fresh doc build each time).

Reply:
I'm seeing some confusing behavior with regard to the meta fields: I'm not able to reproduce the issue with hyphens in the meta fields, and indeed the Python and C++ apigen modules emit `:hide-edit-link:` which is correctly respected.  However, I am seeing that in some cases more than the first meta field is being ignored, but it will need more investigation to figure out what is going on.

For example, after adding `if docname == "index": print(meta)` to `_html_page_context` in `nav_adapt.py`, if I change the start of `docs/index.rst` to:

```
:foo-bar: abc def
:hero: A responsive Material Design theme for Sphinx sites.
```

Then both are present in meta:

```
{'foo-bar': 'abc def', 'hero': 'A responsive Material Design theme for Sphinx sites.'}
```

but if I reverse the order at the top of `docs/index.rst` to:

```
:hero: A responsive Material Design theme for Sphinx sites.
:foo-bar: abc def
```

The only `hero` is present:

```
{'hero': 'A responsive Material Design theme for Sphinx sites.'}
```

Reply:
Interesting, I didn't consider the order when diagnosing. I did try putting blank lines between the metadata in case the lack of a value associated with a metadata key may have been a problem.

Reply:
The only thing I can find in docutils changelogs is that they added a new directive for inserting [`<meta>`](https://docutils.sourceforge.io/docs/ref/doctree.html#meta) elements in v0.18. I also see they stopped stripping multiple `-` from CSS class names (which I remember having to contend with when porting one of upstream features to a theme ext).

Reply:
The problem seems to be related to having an `rst_prolog` specified in `conf.py`: the issue goes away when `rst_prolog` is removed.  It looks like the issue is in docutils but I'm still investigating.

Reply:
Actually the bug is in Sphinx.  When sphinx prepends the prolog, it attempts to insert it after the docinfo, which it detects using a regular expression:

https://github.com/sphinx-doc/sphinx/blob/93215f0a64a2e27656ff2bfe6958a20b1e6b6e1e/sphinx/util/rst.py#L76

https://github.com/sphinx-doc/sphinx/blob/93215f0a64a2e27656ff2bfe6958a20b1e6b6e1e/sphinx/util/rst.py#L26

This regular expression does not correctly match the full syntax allowed for fields for multiple reasons, but in particular it uses "\w" to match the field name, which excludes hyphens.

I think the intent here is to allow `rst_prolog` to insert content into the document (rather than just directives like `.. role: python(code)`), without preventing the document from specifying metadata fields.

For our `rst_prolog`, since it does not generate any docutils nodes, it makes no difference if it is inserted before or after the metadata fields.  The problem occurs if it is inserted in the middle of the metadata fields: since `:hero:` matches `docinfo_re` but `:foo-bar:` does not, the prolog gets inserted after `:hero:`, which results in the metadata fields being parsed as two separate field lists, and docutils only treats the first one as docinfo fields.

Reply:
Sounds like we should propose an amendment to regex in docutils.

Reply:
I filed this bug:
https://github.com/sphinx-doc/sphinx/issues/10932

As a workaround, leaving an initial blank line at the start of file before the docinfo fields mitigates the issue; since the regex doesn't match blank lines, the prolog will be inserted before it, and not split the docinfo fields.

Reply:
Ok I will revert the metadata key names back to using hyphens and test the workaround.